### DISCUSSION TITLE
使用docker部署，无法获取原本文件夹内的图片

### DISCUSSION THREAD

**Discussion by RikinYoung** [hv0905/NekoImageGallery#45] [Q&A]
"/volume1/photo/Photos_Image/A_QQ image:/opt/NekoImageGallery/static"
只能获取上传的图片，不能查看A_QQ image原有的图片。
且新上传的图片均不在A_QQ image文件夹中
虚心请教如何解决

[a comment by an automated account is not shown]

**Comment by RikinYoung** [comment]
使用 python main.py local-index /opt/NekoImageGallery/static 命令建立索引后解决
受教了

**Reply by hv0905** [reply]
@LOifKaze 
不建议将现有目录直接映射到 `/opt/NekoImageGallery/static`，因为该目录是服务端的资源存储目录，服务端会将已索引的图片按照其 ID 存储在此处。如果使用现有目录，可能会导致该目录中存入大量已索引的图片。建议为映射操作使用一个新的空白目录。

在 Docker 容器中，如果你想使用 `local-index` 功能，可以通过类似以下的命令实现：

```
docker run --rm -it -v /path/to/your/static/dir:/opt/NekoImageGallery/static -v /path/to/your/existing/photos/dir:/opt/temp -e <你的环境变量配置> edgeneko/neko-image-gallery:latest local-index /opt/temp
```

即将你的源目录（`/path/to/your/existing/photos/dir`）映射到 `/opt/temp`，然后图片会被自动索引并存储到静态目录（`/path/to/your/static/dir`）中。
